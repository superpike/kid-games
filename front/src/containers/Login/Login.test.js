import React from 'react';
import { render } from '@testing-library/react';
import { Login } from './index';
import '@testing-library/jest-dom/extend-expect';

describe('Testing Login container', () => {
  const fakeLogin = jest.fn(data => data);
  const fakeCancel = jest.fn(data => data);

  xtest('should match to snapshot', () => {
    const { container } = render(<Login />);
    expect(container).toMatchSnapshot();
  });

  describe('Buttons clicks', () => {
    xtest('should login', () => {
      const { getByTestId } = render(
        <Login login={fakeLogin} cancel={fakeCancel} />
      );
      const button = getByTestId('login');
      button.click();
      expect(fakeLogin).toHaveBeenCalledTimes(1);
    });
    xtest('should cancel login', () => {
      const { getByTestId } = render(
        <Login login={fakeLogin} cancel={fakeCancel} />
      );
      const button = getByTestId('cancel');
      button.click();
      expect(fakeCancel).toHaveBeenCalledTimes(1);
    });
  });
});
