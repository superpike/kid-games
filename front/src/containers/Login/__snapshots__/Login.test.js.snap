// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Login container should match to snapshot 1`] = `
<div>
  <div
    class="wrapper"
  >
    Login
  </div>
</div>
`;
